### ISSUE TITLE
[SCM-943] scm:check-local-modification does not support excludes

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#1179]
**[Michiel Kalkman](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michielgkalkman)** opened **[SCM-943](https://issues.apache.org/jira/browse/SCM-943?redirect=false)** and commented

It seems like scm:check-local-modification does not support the excludes and includes parameters. Which is annoying, because I want to exclude *.iml from Intellij from this check.

Looking (but not yet testing) at CheckLocalModificationsMojo::execute(), it does:

```java
result = getScmManager().status( repository, new ScmFileSet( baseDirectory ) );
```

So it does not address neither includes, nor excludes parameter.

My first though would be that something like this is the start of fix:

```java
result = getScmManager().status( repository, new ScmFileSet( getWorkingDirectory(), getIncludes(), getExcludes() ) );
```

Is this something that has not yet been implemented, or is there a user error on my part?
 

---

**Remote Links:**
- [GitHub Pull Request #139
](https://github.com/apache/maven-scm/pull/139)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

While the request ist valid, IDE files belong to the ignore config of your SCM.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [bbd23dcfa9fced2c5a2f91abb47be1c4a660e6ea](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=bbd23dcfa9fced2c5a2f91abb47be1c4a660e6ea) and with [d7107dac448fa0117c35726e5ea52260340f3574](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=d7107dac448fa0117c35726e5ea52260340f3574) for `maven-scm-1.x` branch.
